pkg/datapath/linux/node.go

```diff
@@ -413,6 +417,19 @@ func (n *linuxNodeHandler) nodeUpdate(oldNode, newNode *node.Node) error {
 		n.deleteDirectRoute(newNode.IPv6AllocCIDR, newIP6)
 	}
 
+	if n.nodeConfig.EnableIPSec {
+		new4Net := &net.IPNet{IP: newIP4, Mask: newNode.IPv4AllocCIDR.Mask}
+		new6Net := &net.IPNet{IP: newIP6, Mask: newNode.IPv6AllocCIDR.Mask}
+		n.replaceHostRules()
```

Comment:
maybe obvious we don't need to replaceHostRules on every addition at least :/ thinking...

Reply:
On the plus side this is still gated by `EnableIPSec`, so this could be addressed in a future PR.

Reply:
This code above I believe should only be called "NodeConfigurationChanged is called when the LocalNodeConfiguration has changed" which I think should be the correct context for above code.

But in general I agree with the comment. Currently the code to add the routes to the kernel are listening on the ipcache so any entry in there results in touching the route update kernel side.

Reply:
OK, I think we're on the same page. My comment is perhaps a bit more about the implementation underlying `replaceHostRules` in that it's a "replace everything" rather than making a specific add/delete, and the replace appears to look through potentially many rules.

Reply:
Sure although at the moment there is only two rules so doing a replace on two rules shouldn't hurt much. I expect LocalNodeConfiguration changes are relatively infrequent.